doc/vxlan/Overlay ECMP with BFD.md

```diff
@@ -0,0 +1,216 @@
+
+## 1.1 Functional requirements
+
+At a high level the following should be supported:
+
+- Configure ECMP with Tunnel Nexthops (IPv4 and IPv6)
+- Tunnel Endpoint monitoring via BFD
+- Add/Withdraw Nexthop based on Tunnel or Endpoint health
+
+## 1.2 CLI requirements
+- User should be able to show the BFD session
+- User should be able to show the Vnet routes
+
+## 1.3 Warm Restart requirements
+No special handling for Warm restart support.
+
+# 2 Modules Design
+
+The following are the schema changes. 
+
+## 2.1 Config DB
+
+Existing Vxlan and Vnet tables. 
+
+### 2.1.1 VXLAN Table
+```
+VXLAN_TUNNEL|{{tunnel_name}} 
+    "src_ip": {{ip_address}} 
+    "dst_ip": {{ip_address}} (OPTIONAL)
+```
+### 2.1.2 VNET/Interface Table
+```
+VNET|{{vnet_name}} 
+    "vxlan_tunnel": {{tunnel_name}}
+    "vni": {{vni}} 
+    "scope": {{"default"}} (OPTIONAL)
+    "peer_list": {{vnet_name_list}} (OPTIONAL)
+```
+
+## 2.2 APP DB
+
+### VNET
+
+The following are the changes for Vnet Route table
+
+Existing:
+
+``` 
+VNET_ROUTE_TUNNEL_TABLE:{{vnet_name}}:{{prefix}} 
+    "endpoint": {{ip_address}} 
+    "mac_address":{{mac_address}} (OPTIONAL) 
+    "vni": {{vni}}(OPTIONAL) 
+```
+
+Proposed:
+```
+VNET_ROUTE_TUNNEL_TABLE:{{vnet_name}}:{{prefix}}  
+    "endpoint": {{ip_address1},{ip_address2},...} 
+    "endpoint_monitor": {{ip_address1},{ip_address2},...} (OPTIONAL) 
+    "mac_address":{{mac_address1},{mac_address2},...} (OPTIONAL)  
+    "vni": {{vni1},{vni2},...} (OPTIONAL) 
+    "weight": {{w1},{w2},...} (OPTIONAL) 
+    “profile”: {{profile_name}} (OPTIONAL) 
+```
+
+```
+key                      = VNET_ROUTE_TUNNEL_TABLE:vnet_name:prefix ; Vnet route tunnel table with prefix 
+; field                  = value 
+ENDPOINT                 = list of ipv4 addresses    ; comma separated list of endpoints 
+ENDPOINT_MONITOR         = list of ipv4 addresses    ; comma separated list of endpoints 
+MAC_ADDRESS              = 12HEXDIG                  ; Inner dst mac in encapsulated packet 
+VNI                      = DIGITS                    ; VNI value in encapsulated packet 
+WEIGHT                   = DIGITS                    ; Weights for the nexthops, comma separated (Optional) 
+PROFILE                  = STRING                    ; profile name to be applied for this route, for community  
+                                                       string etc (Optional) 
+```
+
+### BFD
+
+```
+BFD_SESSION:{{ifname}}:{{prefix}}  
+    "tx_interval": {{interval}} (OPTIONAL) 
+    "rx_interval": {{interval}} (OPTIONAL)  
+    "multiplier": {{detection multiplier}} (OPTIONAL) 
+    "shutdown": {{false}} 
+    "multihop": {{false}} 
+    "local_addr": {{ipv4/v6}} (OPTIONAL) 
+    "type": {{string}} (active/passive..) 
+; Defines APP DB schema to initiate BFD session.
+```
+
+## 2.3 Module Interaction
+
+Overlay routes can be programmed via RestAPI or gNMI/gRPC interface which is not described in this document. A highlevel module interaction is shown below
```

Comment:
Do we have support to program vnet routes via REST/gNMI i.e mgmt-framework?

Reply:
Yes for Rest/gNMI. May not be mgmt-framework based